exa/core/data.py

```diff
@@ -169,6 +169,12 @@ def from_tarball(cls, yml=None, qet=None):
             df = pd.read_parquet(qet)
         return cls(df=df, **yml)
 
+    def read_csv(self, *args, **kws):
+        """Convenience method for pandas read_csv"""
+        self.source = pd.read_csv
```

Comment:
I would avoid setting callables likes this as it happens every time this method on this class instance is called. An alternative approach would be:
```
class Data:
    read_csv: callable = pd.read_csv
```

Reply:
I agree this can probably be updated a bit. Rather than assigning `read_csv` as a class attribute as @avmarchenko suggests, we could instead do:
```python
class Data:
    def __init__(self, *args, source=pd.read_csv, **kws):
        self.source = source
```